packages/core/contracts-ovm/insured-bridge/implementation/OVM_BridgeDepositBox.sol

```diff
@@ -1,47 +1,237 @@
 
-contract OVM_BridgeDepositBox is OVM_CrossDomainEnabled {
-    address public l1Owner;
+    address public l1WithdrawContract;
 
-    event Deposit(
+    bool public depositsEnabled = true;
+
+    // Track the total number of deposits. Used as a unique identifier for bridged transfers.
+    uint256 public numberOfDeposits;
+
+    struct WhitelistedToken {
+        address l1Token;
+        uint64 lastBridgeTime;
+    }
+
+    // Mapping of whitelisted L2Token to WhitelistedToken. Contains L1 TokenAddress and the last time this token type
+    // was bridged. Used to rate limit bridging actions to prevent DOS on L1.
+    mapping(address => WhitelistedToken) public whitelistedTokens;
+
+    // Minimum time that must elapse between bridging actions for a given token. Used to rate limit bridging back to L1.
+    uint64 public minimumBridgingDelay;
+
+    /****************************************
+     *                EVENTS                *
+     ****************************************/
+
+    event L1WithdrawContractChanged(address oldL1WithdrawContract, address newL1WithdrawContract);
+    event TokenWhitelisted(address l1Token, address l2Token);
+    event DepositsEnabled(bool enabledResultantState);
+    event L2FundsDeposited(
         uint256 transferId,
         uint256 timestamp,
         address sender,
         address recipient,
-        address originToken,
+        address l1Token,
         uint256 amount,
-        uint256 networkId,
         uint256 maxFee
     );
-    event WithdrawalContractSet(address withdrawalContract);
-    event TokenWhitelisted(address originToken, address destinationToken);
-    event DepositContractEnabledStateToggled(bool enabled);
+    event TokensBridged(address l2Token, uint256 numberOfTokensBridged, uint256 l1Gas, address caller);
+
+    /****************************************
+     *               MODIFIERS              *
+     ****************************************/
 
-    constructor(address _l2CrossDomainMessenger, address _l1Owner) OVM_CrossDomainEnabled(_l2CrossDomainMessenger) {
-        l1Owner = _l1Owner;
+    modifier onlyIfDepositsEnabled() {
+        require(depositsEnabled, "Contract is disabled");
+        _;
     }
 
-    // Admin functions
+    /**
+     * @notice Construct the OVM Bridge Deposit Box
+     * @param _l1WithdrawContract Address of the bridge withdraw contract on L1. Tokens are sent to this address at the
+     *          conclusion of fast relay. This address is the "owner" of the deposit box.
+     * @param _minimumBridgingDelay Minimum second that must elapse between L2->L1 token transfer to prevent dos.
+     * @param timerAddress Timer used to synchronize contract time in testing. Set to 0x000... in production.
+     */
+    constructor(
+        address _l1WithdrawContract,
+        uint64 _minimumBridgingDelay,
+        address timerAddress
+    ) OVM_CrossDomainEnabled(Lib_PredeployAddresses.L2_CROSS_DOMAIN_MESSENGER) OVM_Testable(timerAddress) {
```

Comment:
Should I be using `Lib_PredeployAddresses.L1_CROSS_DOMAIN_MESSENGER` in `WithdrawalBox`?

Reply:
ideally, yes. they'll be the same on all networks and it makes things a bit cleaner imo.